test/unit/spec/common/jobCommChannel-spec.js

```diff
@@ -853,49 +825,18 @@ define([
         });
 
         describe('job mapping', () => {
-            const statusData = ResponseData[jcm.MESSAGE_TYPE.STATUS];
-            // generate the expected mapping
-            const responseDataMapping = {};
-            Object.values(statusData).forEach((job) => {
-                const jobId = job[jcm.PARAM.JOB_ID];
-                if (job.jobState && job.jobState.batch_id) {
-                    responseDataMapping[jobId] = { [jcm.PARAM.BATCH_ID]: job.jobState.batch_id };
-                } else {
-                    responseDataMapping[jobId] = { [jcm.PARAM.JOB_ID]: jobId };
-                }
-            });
-
-            it('should create a job mapping from job updates', () => {
-                const comm = new JobCommChannel(),
-                    msg = makeCommMsg(jcm.MESSAGE_TYPE.STATUS, statusData);
-
-                return comm.initCommChannel().then(() => {
-                    comm.handleCommMessages(msg);
-                    expect(comm._jobMapping).toEqual(responseDataMapping);
-                });
-            });
-
-            it('should create a job mapping from job info data', () => {
-                const infoData = ResponseData[jcm.MESSAGE_TYPE.INFO];
-                const comm = new JobCommChannel(),
-                    msg = makeCommMsg(jcm.MESSAGE_TYPE.INFO, infoData);
-
-                return comm.initCommChannel().then(() => {
-                    comm.handleCommMessages(msg);
-                    expect(comm._jobMapping).toEqual(responseDataMapping);
-                });
-            });
-
-            // since the retry data includes the job status,
-            // we can reconstitute the job mapping from it
-            it('should create a job mapping from job retry data', () => {
-                const retryData = ResponseData[jcm.MESSAGE_TYPE.RETRY];
-                const comm = new JobCommChannel(),
-                    msg = makeCommMsg(jcm.MESSAGE_TYPE.RETRY, retryData);
+            ['STATUS', 'INFO', 'RETRY', 'LOGS'].forEach((type) => {
```

Comment:
All job message types apart from `ERROR` contain the batch ID, so they can be used to create the job mapping. For some reason I did not originally include log messages in the tests... rectified now.